Test/Microsoft.ApplicationInsights.Test/Shared/Extensibility/Implementation/TelemetryProcessorChainBuilderTest.cs

```diff
@@ -72,22 +71,43 @@ public void BuildUsesTelemetryProcesorFactoryOnEachCall()
             var builder2 = new TelemetryProcessorChainBuilder(tc2);
             builder2.Use((next) => new StubTelemetryProcessor(next));
             builder2.Build();
-            
+
             Assert.AreNotSame(tc1.TelemetryProcessors, tc2.TelemetryProcessors);
         }
 
         [TestMethod]
         public void BuildOrdersTelemetryChannelsInOrderOfUseCalls()
         {
-           var config = new TelemetryConfiguration(string.Empty, new StubTelemetryChannel());
-           StringBuilder outputCollector = new StringBuilder();
-           var builder = new TelemetryProcessorChainBuilder(config);
-           builder.Use((next) => new StubTelemetryProcessor(next) { OnProcess = (item) => { outputCollector.Append("processor1"); } });
-           builder.Use((next) => new StubTelemetryProcessor(next) { OnProcess = (item) => { outputCollector.Append("processor2"); } });
-           builder.Build();
-           config.TelemetryProcessorChain.Process(new StubTelemetry());            
-
-           Assert.AreEqual("processor1processor2", outputCollector.ToString());
+            var config = new TelemetryConfiguration(string.Empty, new StubTelemetryChannel());
+            StringBuilder outputCollector = new StringBuilder();
+            var builder = new TelemetryProcessorChainBuilder(config);
+            builder.Use((next) => new StubTelemetryProcessor(next) { OnProcess = (item) => { outputCollector.Append("processor1"); } });
+            builder.Use((next) => new StubTelemetryProcessor(next) { OnProcess = (item) => { outputCollector.Append("processor2"); } });
+            builder.Build();
+            config.TelemetryProcessorChain.Process(new StubTelemetry());
+
+            Assert.AreEqual("processor1processor2", outputCollector.ToString());
+        }
+
+        [TestMethod]
+        public void BuildWillInitializeModules()
+        {
+            var tc1 = new TelemetryConfiguration();
+            var builder1 = new TelemetryProcessorChainBuilder(tc1);
+            builder1.Use((next) => new MockProcessorModule());
+            builder1.Build();
+
+            Assert.AreEqual(2, tc1.TelemetryProcessors.Count); // Transmission is added by default
+            Assert.IsTrue(((MockProcessorModule)tc1.TelemetryProcessors[0]).ModuleInitialized, "Module was not initialized.");
+        }
+
+        private class MockProcessorModule : ITelemetryProcessor, ITelemetryModule
```

Comment:
I know this is common practice inside test classes/files, but I still prefer to see mocks in their own sub-folder of the test. I'll leave it to you to decide whether or not this is important, as I don't think it's commonplace in our codebase as yet.